ligo/swaps.mligo

```diff
@@ -147,8 +147,19 @@ let rec y_to_x_rec (p : y_to_x_rec_param) : y_to_x_rec_param =
 let update_storage_x_to_y (s : storage) (dx : nat) : (nat * nat * storage) =
     let r = x_to_y_rec {s = s ; dx = dx ; dy = 0n} in
     let dx_spent = assert_nat (dx - r.dx, internal_309) in
+    let s_new = {s with balance = {x = s.balance.x + dx_spent ;  y = assert_nat (s.balance.y - r.dy, internal_insufficient_balance_err)}} in
+#if Y_IS_CTEZ
+    (* If Y is CTEZ then the received Y are the total converted ones minus
+    * the protocol fee.
+    * Note: it's important to remove this _after_ updating the storage because
+    * we want to "burn" this fee, in the sense that this contract will continue
+    * to hold these tokens indefinitely, but this amount won't be part of the
+    * stored (and usable) 'balance'.
```

Comment:
Just some economics amateur comment:

Does this mean, that if I perform enough swaps, eventually all the Y tokens will become unusable and so the contract can be thrown away?

On the one hand, the current formula is intuitive and convenient. On the other, it seems a bit counter-intuitive from the following perspective: usually, the fewer tokens are available to people, the more costly they get, but here, even as we get less available tokens - nothing changes.

Reply:
It looks like this PR needs fixing because the `balance` filed in storage is actually unused (see #35 and the discussion on Slack).

However, your questions are still valid, I will push an update to this PR and answer them, please ignore this in its current status.

Reply:
OK, so I pushed the update.

The change with the previous logic is that this is now symmetric, aka the fee will remove the same amount of `dy` regardless of the direction of the swap (all other variables being the same/inverted).

The protocol fee is a percentage of the input CTEZ _left after the swap fee_ or (in the other entrypoint) a percentage of the CTEZ obtained from the conversion.

Reply:
Regarding your question, I went back to the old discussion and it seems that while the decommissioning/"burning" of the CTEZ taken by the protocol fee don't directly change the price the idea is to only take them out of the circulation and let the law of offer and demand do their job.

That is, from the point of view of CTEZ in general (not necessarily on the prices relative to the other token involved in the CFMM).

Reply:
Nice, got it :+1: